src/site/content/en/blog/generic-sensor/index.md

```diff
@@ -0,0 +1,545 @@
+be [polyfilled](#polyfill).
+
+## Available generic sensor APIs {: #available-generic-sensor-apis }
+
+At the time of writing, there are several sensors that you can experiment with.
+
+**Motion sensors:**
+
+- `Accelerometer`
+- `Gyroscope`
+- `LinearAccelerationSensor`
+- `AbsoluteOrientationSensor`
+- `RelativeOrientationSensor`
+- `GravitySensor`
+
+**Environmental sensors:**
+
+- `AmbientLightSensor`
+- `Magnetometer`
+
+## Feature detection {: #feature-detection }
+
+Feature detection of hardware APIs is tricky, since you need to detect both whether the browser
+supports the interface in question, _and_ whether the device has the corresponding sensor. Checking
+whether the browser supports an interface is straightforward. (Replace `Accelerometer` with any of
+the other interfaces mentioned [above](#available-generic-sensor-apis).)
+
+```js
+if ('Accelerometer' in window) {
+  // The `Accelerometer` interface is supported by the browser.
+  // Does the device have an accelerometer, though?
+}
+```
+
+For an actually meaningful feature detection result, you need to try to connect to the sensor, too.
+This example illustrates how to do that.
+
+```js
+let accelerometer = null;
+try {
+  accelerometer = new Accelerometer({ frequency: 10 });
+  accelerometer.addEventListener('error', (event) => {
+    // Handle runtime errors.
+    if (event.error.name === 'NotAllowedError') {
+      console.log('Permission to access sensor was denied.');
+    } else if (event.error.name === 'NotReadableError') {
+      console.log('Cannot connect to the sensor.');
+    }
+  });
+  accelerometer.addEventListener('reading', () => reloadOnShake(accelerometer));
+  accelerometer.start();
+} catch (error) {
+  // Handle construction errors.
+  if (error.name === 'SecurityError') {
+    console.log('Sensor construction was blocked by the Permissions Policy.');
+  } else if (error.name === 'ReferenceError') {
+    console.log('Sensor is not supported by the User Agent.');
+  } else {
+    throw error;
+  }
+}
+```
+
+## Polyfill
+
+For browsers that do not support the Generic Sensor API, a
+[polyfill](https://github.com/kenchris/sensor-polyfills) is available. The polyfill allows you to
+load only the relevant sensors' implementations.
+
+```js
+// Import the objects you need.
+import { Gyroscope, AbsoluteOrientationSensor } from './src/motion-sensors.js';
+
+// And they're ready for use!
+const gyroscope = new Gyroscope({ frequency: 15 });
+const orientation = new AbsoluteOrientationSensor({ frequency: 60 });
+```
+
+## What are all these sensors? How can I use them? {: #what-are-sensors-how-to-use-them }
+
+Sensors is an area that might need a brief introduction. If you are familiar with sensors, you can
+jump right to the [hands-on coding section](#lets-code). Otherwise, let's look at each supported
+sensor in detail.
+
+### Accelerometer and linear acceleration sensor {: #acceleration-and-linear-accelerometer-sensor }
+
+ <figure class="w-figure">
```

Comment:
Shall we specify a width of 100% to avoid mini videos?

Reply:
D'accord. The quality is still tolerable.